src/sqlfluff/core/parser/markers.py

```diff
@@ -188,16 +187,19 @@ def with_working_position(self, line_no: int, line_pos: int):
         )
 
     def is_literal(self) -> bool:
-        """Infer literalness from context."""
-        # NOTE: We should define what the implication of is_literal is here.
-        # TODO: I think we were inconsistent in whether a zero length source slice
-        # is (or is not) literal. The lexer initially defined it as False, this
-        # function will say True. Not sure what the implications of that are yet.
-        # NOTE: So far I can see it only controls:
-        # 1. The patch_type of a FixPatch, which appears to then never be referred to.
-        # 2. The ignoring of templated sections in `remove_templated_errors`.
-        # 3. Whether we can return simply in `iter_patches`.
-        # 4. The application of L046 which is looking for tags.
+        """Infer literalness from context.
+
+        is_literal should return True if a fix can be applied across this area
+        without concern. This obviously applies to any slices which are the same
```

Comment:
Can you clarify what you mean by "without concern"? Concern that ______? If you could, please consider providing an example or two, also a counterexample.

Reply:
Yeah - not very clear is it. I've made it clearer.